### PR TITLE
[opencc] allow to build on arm

### PR DESCRIPTION
**Pull request by shenlebantongying**
This package can be built on ARM (I tested on macOS) today.

Homebrew has been providing ARM build of this package for a very long time https://formulae.brew.sh/formula/opencc

Their CI also has Arm build since 3 years ago https://github.com/BYVoid/OpenCC/blame/e5d6c5f1b78e28a5797e7ad3ede3513314e544b7/.travis.yml#L16-L18

### CODE REVIEW

File: ports/opencc/vcpkg.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "description": "A project for conversions between Traditional Chinese, Simplified Chinese and Japanese Kanji (Shinjitai)",
```
Comment (jimwang118) [inline ports/opencc/vcpkg.json:5]: "port-version": 2,

Comment (jimwang118) [reply ports/opencc/vcpkg.json:5]: Run the command: ./vcpkg x-add-version {portname}, and then submit the generated result.

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/opencc/vcpkg.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
...
   "homepage": "https://github.com/BYVoid/OpenCC",
   "license": "Apache-2.0",
-  "supports": "!(arm | uwp)",
+  "supports": "!(uwp)",
```
Comment (dg0yt) [inline ports/opencc/vcpkg.json:8]:
```suggestion
  "supports": "!uwp",
```

**Review by dg0yt — COMMENTED** [review]
The real problem was never arm, but insufficient cross-build support in the vcpkg port. That's why arm android fails now: It tries to run  `opencc_dict` on the host. But in vcpkg, ports may build only for one triplet, and everything else must be modeled via host dependencies.

File: ports/opencc/vcpkg.json (hunk 2 above)
Comment (shenlebantongying) [reply ports/opencc/vcpkg.json:8]: So, should I also add `!android` and `!x86` to pass CI?

I don't think upstream is going to fix them :sweat_smile:

Comment (dg0yt) [reply ports/opencc/vcpkg.json:8]: No, it is not an upstream issue. It is how to adapt the build to vcpkg's concept of separating host triplet and target triplet.

**Comment by shenlebantongying** [comment]
@microsoft-github-policy-service agree

**Review by jimwang118 — DISMISSED** [review]

**Comment by BillyONeal** [comment]
@shenlebantongying please fix the build failures and then mark ready for review. I'm guessing you just need a "!android" in supports?

**Pull request closed without merging**